custom_code/hooks.py

```diff
@@ -17,6 +17,34 @@
 logger = logging.getLogger(__name__)
 
 
+def process_reduced_ztf_data(target, candidates):
+    """Ingest data from the ZTF JSON format into ``ReducedDatum`` objects. Mostly copied from tom_base v2.13.0."""
+    for candidate in candidates:
+        if all([key in candidate['candidate'] for key in ['jd', 'magpsf', 'fid', 'sigmapsf']]):
+            nondetection = False
+        elif all(key in candidate['candidate'] for key in ['jd', 'diffmaglim', 'fid']):
+            nondetection = True
+        else:
+            continue
+        jd = Time(candidate['candidate']['jd'], format='jd', scale='utc')
+        jd.to_datetime(timezone=TimezoneInfo())
+        value = {
+            'filter': {1: 'g', 2: 'r', 3: 'i'}[candidate['candidate']['fid']]
+        }
+        if nondetection:
+            value['limit'] = candidate['candidate']['diffmaglim']
+        else:
+            value['magnitude'] = candidate['candidate']['magpsf']
+            value['error'] = candidate['candidate']['sigmapsf']
+        rd, _ = ReducedDatum.objects.get_or_create(
+            timestamp=jd.to_datetime(timezone=TimezoneInfo()),
+            value=value,
+            source_name='ZTF',
+            source_location=alert['zid'],
```

Comment:
this was breaking so I edited alert --> candidate

Reply:
fixed